Gem/Code/Source/Components/NetworkTeleportComponent.cpp

```diff
@@ -0,0 +1,153 @@
+						AZ_CRC_CE("Game"))
+					->DataElement(0, &NetworkTeleportComponent::m_reset,
+						"Destination", "Entity at which to teleport the colliding player")
+					;
+			}
+		}
+	}
+
+	NetworkTeleportComponent::NetworkTeleportComponent()
+		: m_trigger([this](
+			AzPhysics::SceneHandle,
+			const AzPhysics::TriggerEventList& eventList)
+			{
+				OnTriggerEvents(eventList);
+			})
+	{
+	}
+
+	void NetworkTeleportComponent::Activate()
+	{
+		auto* si = AZ::Interface<AzPhysics::SceneInterface>::Get();
+		if (si != nullptr)
+		{
+			AzPhysics::SceneHandle sh = si->GetSceneHandle(
+				AzPhysics::DefaultPhysicsSceneName);
+			si->RegisterSceneTriggersEventHandler(sh, m_trigger);
+		}
+	}
+
+	void NetworkTeleportComponent::OnTriggerEvents(
+		const AzPhysics::TriggerEventList& eventList)
+	{
+		for (const AzPhysics::TriggerEvent& event : eventList)
+		{
+			if (!event.m_triggerBody || (event.m_triggerBody->GetEntityId() != GetEntityId()))
+			{
+				continue;
+			}
+
+			if (event.m_type == AzPhysics::TriggerEvent::Type::Enter)
+			{
+				AZ_Printf("Teleporter", "physics event: %s\n", "Enter");
+
+				if (event.m_otherBody)
+				{
+					// TODO: validate it's a player entity
+					AZ::Vector3 teleportLocation = GetDestination();
+					AZ_Printf("Teleporter", "destination point X is: %f\n",
+						teleportLocation.GetX());
+
+					AZ::Entity* otherEntity = GetCollidingEntity(event.m_otherBody);
+					TeleportPlayer(teleportLocation, otherEntity);
+				}
+			}
+		}
+	}
+
+	AZ::Entity* NetworkTeleportComponent::GetCollidingEntity(AzPhysics::SimulatedBody* collidingBody) const
+	{
+		AZ::Entity* collidingEntity = nullptr;		
+		if (collidingBody)
+		{
+			AZ::EntityId collidingEntityId = collidingBody->GetEntityId();
+			AZ::ComponentApplicationBus::BroadcastResult(
+				collidingEntity, &AZ::ComponentApplicationBus::Events::FindEntity, collidingEntityId);
+		}
+		return collidingEntity;
+	}
+
+	AZ::Vector3 NetworkTeleportComponent::GetDestination() const
+	{
+		AZ::Vector3 newLocation = AZ::Vector3::CreateZero();
+		AZ::TransformBus::EventResult(newLocation, m_reset,
+			&AZ::TransformBus::Events::GetWorldTranslation);
+
+		return newLocation;
+	}
+
+	void NetworkTeleportComponent::TeleportPlayer(
+		const AZ::Vector3& v, AZ::Entity* entity)
+	{
+		if (entity)
+		{
+			AZ::EntityId playerToTeleport = entity->GetId();
+
+			AZ_Printf(
+				"Teleporter", "attempting to fling entity: %s\n", "PLAYER");
+
+			// disable physics so we can move entity
+			Physics::RigidBodyRequestBus::Event(playerToTeleport,
+				&Physics::RigidBodyRequestBus::Events::DisablePhysics);
+
+			// move the entity
+			AZ::TransformBus::Event(playerToTeleport,
+				&AZ::TransformBus::Events::SetWorldTranslation, v);
+
+			// increment resetCount so it doesn't try to interplate from last location
+			Multiplayer::NetworkTransformComponent* netTransform =
+				entity->FindComponent<Multiplayer::NetworkTransformComponent>();
```

Comment:
When you do network this component you should NetworkTransformComponent a dependency in the xml definition. This will generate an accessor for  NetworkTransformComponent from here that's faster than doing a FindComponent.

Reply:
It specifically allows a NetworkTeleportComponent access to a NetworkTransformComponent on the same entity.